src/components/Editor/SearchBar.js

```diff
@@ -149,152 +127,48 @@ class SearchBar extends Component {
     this.closeSearch(e);
   }
 
-  clearSearch() {
-    const { editor: ed, query, modifiers } = this.props;
-    if (ed && modifiers) {
-      const ctx = { ed, cm: ed.codeMirror };
-      removeOverlay(ctx, query, modifiers.toJS());
-    }
-  }
-
-  closeSearch(e: SyntheticEvent) {
-    const { editor, setFileSearchQuery, searchOn } = this.props;
-
-    if (editor && searchOn) {
-      setFileSearchQuery("");
-      this.clearSearch();
-      this.props.setActiveSearch();
-      this.props.clearHighlightLineRange();
-      e.stopPropagation();
-      e.preventDefault();
-    }
-  }
-
-  toggleSearch(e: SyntheticKeyboardEvent) {
+  traverseResults(e: SyntheticKeyboardEvent, dir: boolean) {
     e.stopPropagation();
     e.preventDefault();
-    const { editor } = this.props;
-
-    if (!this.props.searchOn) {
-      this.props.setActiveSearch("file");
-    }
-
-    if (this.props.searchOn && editor) {
-      const selection = editor.codeMirror.getSelection();
-      this.setSearchValue(selection);
-      if (selection !== "") {
-        this.doSearch(selection);
-      }
-      this.selectSearchInput();
-    }
-  }
-
-  setSearchValue(value: string) {
-    const searchInput = this.searchInput();
-    if (value == "" || !searchInput) {
+    if (!this.props.editor) {
```

Comment:
moving this below `e.stopPropagation` and` e.preventDefault` is what was causing the escape button bug.